server/api.go

```diff
@@ -0,0 +1,243 @@
+	"github.com/mattermost/mattermost-server/v5/plugin"
+)
+
+const externalAPICacheTTL = 3600000
+
+var externalAPICache []byte
+var externalAPILastUpdate int64
+var externalAPICacheMutex sync.Mutex
+
+type StartMeetingRequest struct {
+	ChannelID string `json:"channel_id"`
+	Personal  bool   `json:"personal"`
+	Topic     string `json:"topic"`
+	MeetingID int    `json:"meeting_id"`
+}
+
+type StartMeetingFromAction struct {
+	model.PostActionIntegrationRequest
+	Context struct {
+		MeetingID    string `json:"meeting_id"`
+		MeetingTopic string `json:"meeting_topic"`
+		Personal     bool   `json:"personal"`
+	} `json:"context"`
+}
+
+func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Request) {
+	switch path := r.URL.Path; path {
+	case "/api/v1/meetings/enrich":
+		p.handleEnrichMeetingJwt(w, r)
+	case "/api/v1/meetings":
+		p.handleStartMeeting(w, r)
+	case "/api/v1/config":
+		p.handleConfig(w, r)
+	case "/jitsi_meet_external_api.js":
+		p.proxyExternalAPIjs(w, r)
+	default:
+		http.NotFound(w, r)
+	}
+}
+
+func (p *Plugin) handleConfig(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-Id")
+
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	config, err := p.getUserConfig(userID)
+	if err != nil {
+		http.Error(w, "Internal error", http.StatusInternalServerError)
+		return
+	}
+
+	b, err2 := json.Marshal(config)
+	if err2 != nil {
+		log.Printf("Error marshaling the Config to json: %v", err2)
+		http.Error(w, "Internal error", http.StatusInternalServerError)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	_, _ = w.Write(b)
+}
+
+func (p *Plugin) proxyExternalAPIjs(w http.ResponseWriter, r *http.Request) {
+	externalAPICacheMutex.Lock()
+	defer externalAPICacheMutex.Unlock()
+
+	if externalAPICache != nil && externalAPILastUpdate > (model.GetMillis()-externalAPICacheTTL) {
+		w.Header().Set("Content-Type", "application/javascript")
+		_, _ = w.Write(externalAPICache)
+		return
+	}
+	resp, err := http.Get(p.getConfiguration().JitsiURL + "/external_api.js")
+	if err != nil {
+		http.Error(w, "Internal error", http.StatusInternalServerError)
+		return
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		http.Error(w, "Internal error", http.StatusInternalServerError)
+		return
+	}
+	externalAPICache = body
+	externalAPILastUpdate = model.GetMillis()
+	w.Header().Set("Content-Type", "application/javascript")
+	_, _ = w.Write(body)
+}
+
+func (p *Plugin) handleStartMeeting(w http.ResponseWriter, r *http.Request) {
+	if err := p.getConfiguration().IsValid(); err != nil {
+		http.Error(w, err.Error(), http.StatusTeapot)
+		return
+	}
+
+	userID := r.Header.Get("Mattermost-User-Id")
+
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	user, appErr := p.API.GetUser(userID)
+	if appErr != nil {
+		http.Error(w, "Forbidden", http.StatusForbidden)
```

Comment:
:+1: to the error logging. I don't think we need to handle in other way, if your user id is not a valid user id, you can't access to it. I think is behaving correctly. What would be the suggestion about "handling it a bit more"?

Reply:
I am not 100% sure, but I guess that `p.API.GetUser(...)` may throw an error for different reasons. If the error is about not finding the user, we understand that the userID is not valid, and therefore is a forbidden error and there is no need to log anything. If there is some other error (database error or connection error perhaps?) it would be a InternalServerError and some log should be put in place.
I am 0/5 on whether this is worth it or not make this distinction, but I wanted to raise the possibility.

Reply:
I'm not sure either, I added a log there, I think should be enough for now.